base/deepcopy.jl

```diff
@@ -23,12 +23,12 @@ function deepcopy_internal(x::SimpleVector, stackdict::ObjectIdDict)
 end
 
 function deepcopy_internal(x::ANY, stackdict::ObjectIdDict)
-    if haskey(stackdict, x)
-        return stackdict[x]
-    end
     T = typeof(x)::DataType
     nf = nfields(T)
     (isbits(T) || nf == 0) && return x
+    if haskey(stackdict, x)
+        return stackdict[x]
+    end
```

Comment:
It's much less important than the rest of the change, but checking for `isbits` is faster than doing a dictionary lookup, plus isbits values are never added to the dictionary, so this lets us skip the lookup entirely in those cases.